storage/series/series.go

```diff
@@ -570,7 +570,8 @@ func (s *dbSeries) Close() {
 
 func (s *dbSeries) Reset(id ts.ID) {
 	s.Lock()
-	s.id = ts.BinaryID(id.Data())
+	s.id.Close()
```

Comment:
The pattern has been to put all the closing/returning logic in the `Close(...)` methods on entities.  That way they return them to the pool before they themselves are reused.  

Would be good to move `s.id.Close()` to inside the `Close()` method to follow suit.